### PR TITLE
Move core types to extra package

### PR DESCRIPTION
**Pull request by jkoenig134**
# Readiness checklist

-   [ ] I added/updated tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

# Description / Motivation

As the Core* types were placed in transport before we could not re-use them in e.g. the connector sdk. With an extra package that transport and content can depend on we now have the ability to re-use content types.

### CODE REVIEW

File: packages/transport/src/core/CoreIdHelper.ts
Comment (Milena-Czierlinski) [inline packages/transport/src/core/CoreIdHelper.ts:1]: Why is this not in the core-types package?

File: packages/app-runtime/src/multiAccount/data/LocalAccount.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { serialize, type, validate } from "@js-soft/ts-serval";
-import { CoreAddress, CoreDate, CoreId, CoreSerializable, ICoreDate, ICoreSerializable } from "@nmshd/transport";
+import { ISerializable, Serializable, serialize, type, validate } from "@js-soft/ts-serval";
+import { CoreAddress, CoreDate, CoreId, ICoreDate } from "@nmshd/core-types";
 
-export interface ILocalAccount extends ICoreSerializable {
+export interface ILocalAccount extends ISerializable {
```
Comment (Milena-Czierlinski) [inline packages/app-runtime/src/multiAccount/data/LocalAccount.ts:4]: Why did you change this in some places?

File: packages/consumption/src/modules/attributes/local/LocalAttribute.ts [hunk 2]
```diff
@@ -8,10 +8,11 @@ import {
     RelationshipAttribute,
     RelationshipAttributeJSON
 } from "@nmshd/content";
-import { CoreAddress, CoreDate, CoreId, CoreSynchronizable, ICoreDate, ICoreId, ICoreSynchronizable } from "@nmshd/transport";
+import { CoreAddress, CoreDate, CoreId, ICoreDate, ICoreId } from "@nmshd/core-types";
+import { CoreSynchronizable, ICoreSynchronizable } from "@nmshd/transport";
```
Comment (Milena-Czierlinski) [inline packages/consumption/src/modules/attributes/local/LocalAttribute.ts:12]: Can CoreSynchronizable (and CoreSerializable) also move to core-types?

**Review by Milena-Czierlinski — COMMENTED** [review]
Do the (app-)runtime tsconfig files need `"references": [{ "path": "../core-types" }]` as well?

File: packages/consumption/src/modules/attributes/local/LocalAttribute.ts (hunk 2 above)
Comment (jkoenig134) [reply packages/consumption/src/modules/attributes/local/LocalAttribute.ts:12]: CoreSerializable is a utility type that would bring crypto to the core-types library which we dont want. The long-running plan from my side is to remove this type completely. This will introduce more changes so I will do this in a next iteration.

CoreSynchronizable is a data type strictly coupeled to how the transport sync works so I have not moved this.

File: packages/app-runtime/src/multiAccount/data/LocalAccount.ts (hunk 1 above)
Comment (jkoenig134) [reply packages/app-runtime/src/multiAccount/data/LocalAccount.ts:4]: explained in other comment

File: packages/transport/src/core/CoreIdHelper.ts
Comment (jkoenig134) [reply packages/transport/src/core/CoreIdHelper.ts:1]: because it would **kill** the whole idea of the core-types package. CoreIdHelper is no type and would bring crypto to core-types.

**Comment by jkoenig134** [comment]
No, things are already referenced in library below.

File: packages/core-types/src/CoreDate.ts [hunk 3]
```diff
@@ -138,17 +136,17 @@ export class CoreDate extends CoreSerializable {
     }
 
     private get asValidDateTime(): DateTime<true> {
-        if (!this.dateTime.isValid) throw new TransportError("The date is invalid.");
+        if (!this.dateTime.isValid) throw new Error("The date is invalid.");
```
Comment (Milena-Czierlinski) [inline packages/core-types/src/CoreDate.ts:139]: Would it make sense to have a CoreError here?

Comment (jkoenig134) [reply packages/core-types/src/CoreDate.ts:139]: no, I had this, but then removed it.

File: packages/consumption/src/modules/attributes/local/LocalAttribute.ts (hunk 2 above)
Comment (jkoenig134) [reply packages/consumption/src/modules/attributes/local/LocalAttribute.ts:12]: To not have more confusion I remove CoreSerializable completely.

File: packages/core-types/src/CoreDate.ts (hunk 3 above)
Comment (jkoenig134) [reply packages/core-types/src/CoreDate.ts:139]: To give more context. We already have a CoreError, but this has code, message, ... - this would be too much for this class.

**Review by Milena-Czierlinski — APPROVED** [review]
